src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+namespace Microsoft.Rest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    /// <summary>
+    /// Abstract base class for types that act like enums, but can be extended with arbitrary string values.
+    /// </summary>
+    public abstract class ExtensibleEnum<T> : IEquatable<T> where T : ExtensibleEnum<T>
+    {
+        private static readonly Lazy<Dictionary<string, T>> _nameMap =
+            new Lazy<Dictionary<string, T>>(CreateNameMap, isThreadSafe: true);
+
+        private string _name;
```

Comment:
`name` should be called `value` or `serializedValue` -- `name` is too ambigious

Reply:
👍 

The name `_name` was a holdover from before I generalized this pattern into a base class. Back then it was `AnalyzerName`, which was search-specific, and really was all about names.